lib/eslint-patch.js

```diff
@@ -1,33 +1,48 @@
-'use strict';
-var meld = require('meld');
-var docs = require('./docs');
-var Config = require("eslint/lib/config");
-var ConfigUpgrader = require('./config_upgrader');
-
-var supportedPlugins = ['react', 'babel'];
-
-module.exports = function patcher(eslint) {
-
-  meld.around(eslint.CLIEngine, 'loadPlugins', function(joinPoint) {
-    var pluginNames = joinPoint.args[0];
-    var filteredPluginNames = pluginNames.filter(function(pluginName) {
-      return supportedPlugins.indexOf(pluginName) >= 0;
-    });
-    return joinPoint.proceed(filteredPluginNames);
-  });
-
-  meld.around(eslint.CLIEngine, 'addPlugin', function() {
-    return;
-  });
-
-  // meld.around(eslint.CLIEngine.Config, 'loadPackage', function(joinPoint) {
-  //   var filePath = joinPoint.args[0];
-  //   if (filePath.match(/^eslint-config-airbnb.*/)) {
-  //     return joinPoint.proceed();
-  //   } else {
-  //     return {};
-  //   }
-  // });
+"use strict";
+
+const Plugins = require("eslint/lib/config/plugins");
+const ModuleResolver = require("eslint/lib/util/module-resolver");
+
+const ConfigFile = require("eslint/lib/config/config-file");
+
+const Config = require("eslint/lib/config");
+const ConfigUpgrader = require("./config_upgrader");
+
+module.exports = function patch() {
+  const skippedPlugins = [];
+
+  const resolve = ModuleResolver.prototype.resolve;
+  ModuleResolver.prototype.resolve = function(name, path) {
+    try {
+      return resolve.apply(this, [name, path]);
+    } catch(e) {
+      const pluginMatch = name.match(/eslint-plugin-(.+)/);
+      if(pluginMatch && skippedPlugins.indexOf(pluginMatch[1]) >= 0) {
+        return `${__dirname}/empty-plugin.js`;
+      }
+
+      throw e;
```

Comment:
@maxjacobson Keeping track of skipped plugins, now. If an `extends` config is related to one of them, we also skip it here. `bogus` entry should still result in error, which is raised and propagated all the way.